src/buttons.ts

```diff
@@ -88,272 +107,284 @@ class FileButtons extends Widget {
+   * Create a new untitled file.
    */
   export
-  function createNewItemMenu(widget: FileButtons): Menu {
-    // Create the "new" menu.
-    let handler = (item: MenuItem) => {
-      let type = item.text.toLowerCase();
-      if (type === 'text file') type = 'file';
-      if (type === 'folder') type = 'directory';
-      widget.model.newUntitled(type).then(contents => {
-        if (type !== 'directory') widget.openRequested.emit(contents);
-        widget.model.refresh();
-      },
-      error =>
-        utils.showErrorMessage(widget, 'New File Error', error)
-      );
-    };
-    let items: MenuItem[] = [
+  function createNewFile(widget: FileButtons): void {
+    widget.model.newUntitled('file').then(contents => {
+      widget.openRequested.emit(contents);
+      widget.model.refresh();
+    }).catch(error => {
+      utils.showErrorMessage(widget, 'New File Error', error);
+    });
+  }
+
+  /**
+   * Create a new untitled folder.
+   */
+  export
+  function createNewFolder(widget: FileButtons): void {
+    widget.model.newUntitled('directory').then(contents => {
+      widget.model.refresh();
+    }).catch(error => {
+      utils.showErrorMessage(widget, 'New Folder Error', error);
+    });
+  }
+
+  /**
+   * Create a new untitled notebook.
+   */
+  export
+  function createNewNotebook(widget: FileButtons, spec: IKernelSpecId): void {
+    widget.model.newUntitled('notebook').then(contents => {
+      let started = widget.model.startSession(contents.path, spec.name);
+      return started.then(() => contents);
+    }).then(contents => {
+      widget.openRequested.emit(contents);
+      widget.model.refresh();
+    }).catch(error => {
+      utils.showErrorMessage(widget, 'New Notebook Error', error);
+    });
+  }
+
+  /**
+   * Create a new dropdown menu for the create new button.
+   */
+  export
+  function createDropdownMenu(widget: FileButtons): Menu {
+    let items = [
       new MenuItem({
         text: 'Text File',
-        handler: handler,
+        handler: () => { createNewFile(widget); }
       }),
       new MenuItem({
         text: 'Folder',
-        handler: handler,
+        handler: () => { createNewFolder(widget); }
       }),
       new MenuItem({
         type: MenuItem.Separator
-      }),
-      new MenuItem({
-        text: 'Notebooks',
-        disabled: true
-      }),
-    ]
-    for (var spec of widget.model.kernelSpecs) {
-      items.push(new MenuItem({
-        text: spec.spec.display_name,
-        handler: () => {
-          widget.model.newUntitled('notebook').then(contents => {
-            widget.model.startSession(contents.path, spec.name).then(() => {
-              widget.openRequested.emit(contents);
-              widget.model.refresh();
-            });
-          });
-        }
-      }));
-    }
-    return new Menu(items);
+      })
+    ];
+    let extra = widget.model.kernelSpecs.map(spec => {
+      return new MenuItem({
+        text: `Notebook - ${spec.spec.display_name}`,
```

Comment:
I just did this for now because we don't have a way to put text on a menu separator. Thought it looked better than a disabled item.